### PR TITLE
Manpages

### PR DESCRIPTION
**Pull request by davidmoreno**
This adds manpages for the three binaries that onion generates.

### CODE REVIEW

File: manpages/onion-crl.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+stateless, this effectively creates a live C programming environment where the
+programmer changes some C or HTML code and reloads the page to see the changes.
+
+If the server is stateful, onion-crl is not recommended.
+
+onion-crl assumes a cmake environment where its running into a `build` directory,
+separated from the source, and with the `make` as compilation program.
+
+# ENVIRONMENT
+onin-crl tries its best to find all source files, but if its not working
```
Comment (zacharygrafton) [inline manpages/onion-crl.md]: Suggested wording: onion-crl tries to find all the source files, but if unsuccessful these environment variables can affect the behavior

File: manpages/otemplate.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Refer to Django documentation on use cases and possibilities. Please take into
+account that otemplate just reimplements the most useful functionalities of
+Django tags. If more is needed, please create an issue at Onion HTTP Library
+issue tracker at Github.
+
+# Usage from code using Onion HTTP library
+
+The generated functions should be called with a onion_dict as first function. Normal
+should is something like this:
```
Comment (zacharygrafton) [inline manpages/otemplate.md]: Suggested wording: The generated functions should be called with a onion_dict as the first argument. Normal usage should look like this:

**Comment by zacharygrafton** [comment]
David, I like man pages to be completely honest. I'm wondering if we should just use doxygen to generate man pages for all the documentation? It should be easy enough to add into the cmake files.

**Comment by davidmoreno** [comment]
Sounds cool. I don't know how complete they would be, but we can try.

I think there is some focused option, so on cmake should be only install
instructions.. I will try to have a look later.

David
On 24 Apr 2016 15:01, "Zachary Grafton" <notifications@github.com> wrote:

David, I like man pages to be completely honest. I'm wondering if we should
just use doxygen to generate man pages for all the documentation? It should
be easy enough to add into the cmake files.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/davidmoreno/onion/pull/176#issuecomment-213953429>

**Comment by davidmoreno** [comment]
I just made that suggested changes on the current man pages.

I also checked the mapages generated by doxygen and I think that after some minimal processing we could install many onion_[header].3 and links from onion_[function].3 . 

It can not be used directly as it generates really a lot of not useful manpages, as internal functions (write.3.... from listen_point.c.. I think the field in the struct). Also the normal name for the header manpages is without the onion prefix, so it could interact with other manpages.

Actually the pages themselves are not super manpages, but it's a beginning. To generate them just run doxygen, and they are generated at doc/man/.

David.

**Comment by davidmoreno** [comment]
Finally I got no time to fix the manual page per function. I add the issue to fix for a later release.

**Pull request merged**